### PR TITLE
Added feature import supplyPrice from CSV

### PR DESCRIPTION
**Pull request by chronos25**
@with-shrey  ! I've done the changes for feature import supplyPrice from csv kindly confirm and recheck.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -32,8 +32,8 @@ AWS_S3_REORDER_POINTS_MULTIPLIERS_BUCKET=
 AWS_SQS_ACCESS_KEY_ID=stockup
```
Comment (with-shrey) [inline .env.example:32]: Add minio default clientId secret and bucket names in this file

File: common/models/order-config-model.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ module.exports = function (OrderConfigModel) {
         OrderConfigModel.fetchFileImportHeaders = function (id, options) {
             var importableHeaders = {
                 Inventory: ['orderQuantity', 'fulfilledQuantity'],
-                Product: ['name', 'sku'],
+                Product: ['name', 'sku','supplyprice'],
```
Comment (with-shrey) [inline common/models/order-config-model.js:15]: rename `supplyprice` to be in camelCase similar to other keys 

And maybe move them to an existing constants file

File: common/models/report-model.js [hunk 3]
```diff
@@ -1226,9 +1226,13 @@ module.exports = function (ReportModel) {
                     apiVersion: '2006-03-01',
                     region: ReportModel.app.get('awsS3Region'),
                     accessKeyId: ReportModel.app.get('awsAccessKeyId'),
-                    secretAccessKey: ReportModel.app.get('awsSecretAccessKey')
+                    secretAccessKey: ReportModel.app.get('awsSecretAccessKey'),
+                    endpoint: process.env.ENDPOINT,
```
Comment (with-shrey) [inline common/models/report-model.js:1230]: `ENDPOINT` is very vague 
change it to `S3_ENDPOINT` or some thing that makes more sense 

Also this env variable is missing in .env.example file

File: common/models/report-model.js [hunk 4]
```diff
@@ -1226,9 +1226,13 @@ module.exports = function (ReportModel) {
...
+                    s3ForcePathStyle: true, // needed with minio?
+                    signatureVersion: 'v4'
                 });
-                let s3Bucket = ReportModel.app.get('awsS3CSVImportsBucket');
+                //let s3Bucket = ReportModel.app.get('awsS3CSVImportsBucket');
```
Comment (with-shrey) [inline common/models/report-model.js:1234]: remove commented code

File: worker2.env.example [hunk 5]
```diff
@@ -2,12 +2,15 @@
 # Warehouse Workers fetch jobs/payloads from a queue
 # There are many technologies that can provide a queue implementation: SQS, RabbitMQ, redis, etc.
 ##
+AWS_ACCESS_KEY_ID= <YOUR>
```
Comment (with-shrey) [inline worker2.env.example:5]: change key & secret to minio default key & secret

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js [hunk 6]
```diff
@@ -252,7 +252,10 @@ function readSpreadSheetFromS3(s3Bucket, s3BucketKey, messageId) {
         apiVersion: '2006-03-01',
         region: process.env.AWS_S3_REGION,
         accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+        endpoint: 'http://minio.stockup.localdomain:9000',
+        s3ForcePathStyle: true, // needed with minio?
+          signatureVersion: 'v4'
```
Comment (with-shrey) [inline workers/workers-v2/import-stock-order/import-stock-order-vend.js:258]: incorrect indent/spacing

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js [hunk 7]
```diff
@@ -252,7 +252,10 @@ function readSpreadSheetFromS3(s3Bucket, s3BucketKey, messageId) {
         apiVersion: '2006-03-01',
         region: process.env.AWS_S3_REGION,
         accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+        endpoint: 'http://minio.stockup.localdomain:9000',
```
Comment (with-shrey) [inline workers/workers-v2/import-stock-order/import-stock-order-vend.js:256]: move this value to worker's env file

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js [hunk 8]
```diff
@@ -320,14 +323,15 @@ function mapSpreadSheetDataToOrders(db, orderConfigModel, spreadSheetRows, userM
...
+    let supplyPriceFileHeader = _.findWhere(orderConfigModel.mappings.Product, {stockupHeader: 'supplyprice'}).fileHeader;
     let orderQuantityFileHeader = _.findWhere(orderConfigModel.mappings.Inventory, {stockupHeader: 'orderQuantity'}).fileHeader;
     let fulfilledQuantityFileHeader = _.findWhere(orderConfigModel.mappings.Inventory, {stockupHeader: 'fulfilledQuantity'}).fileHeader;
     let skusToAdd = _.uniq(_.pluck(spreadSheetRows, skuFileHeader));
     //just to make sure to st\ringify if any sku contains just numbers
     let skusToAddStringified = _.map(skusToAdd, function (eachSku) {
         return typeof eachSku === 'string' ? eachSku : JSON.stringify(eachSku);
     });
-    let productModelInstances, storeModelInstances;
+    let supplypriceproductModelInstances, storeModelInstances;
```
Comment (with-shrey) [inline workers/workers-v2/import-stock-order/import-stock-order-vend.js:334]: use camel case for variables

i thunk this variable is not being used

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js [hunk 9]
```diff
@@ -400,7 +403,7 @@ function mapSpreadSheetDataToOrders(db, orderConfigModel, spreadSheetRows, userM
                         eachSpreadSheetRow.supplierModelId = productFound.supplierModelId;
                         eachSpreadSheetRow.categoryModelId = productFound.categoryModelId;
                         eachSpreadSheetRow.categoryModelName = productFound.categoryModel && productFound.categoryModel.length ? productFound.categoryModel[0].name : 'No Category';
-                        eachSpreadSheetRow.supplyPrice = productFound.supply_price;
+                        eachSpreadSheetRow.supplyPrice = eachSpreadSheetRow.supply_price;
```
Comment (with-shrey) [inline workers/workers-v2/import-stock-order/import-stock-order-vend.js:406]: i think this line is useless now ?

**Review by with-shrey — CHANGES_REQUESTED** [review]
Please make the changes and let me know 

I was unable to test because of missing env variables

File: .env.example (hunk 1 above)
Comment (chronos25) [reply .env.example:32]: Done !

File: common/models/order-config-model.js (hunk 2 above)
Comment (chronos25) [reply common/models/order-config-model.js:15]: done !

File: common/models/report-model.js (hunk 3 above)
Comment (chronos25) [reply common/models/report-model.js:1230]: Done

File: common/models/report-model.js (hunk 4 above)
Comment (chronos25) [reply common/models/report-model.js:1234]: Done!

File: worker2.env.example (hunk 5 above)
Comment (chronos25) [reply worker2.env.example:5]: done !

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js (hunk 6 above)
Comment (chronos25) [reply workers/workers-v2/import-stock-order/import-stock-order-vend.js:258]: done !

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js (hunk 8 above)
Comment (chronos25) [reply workers/workers-v2/import-stock-order/import-stock-order-vend.js:334]: Done ! there is no need to remove this variable

File: workers/workers-v2/import-stock-order/import-stock-order-vend.js (hunk 9 above)
Comment (chronos25) [reply workers/workers-v2/import-stock-order/import-stock-order-vend.js:406]: Yes ! All the mentioned changes are completed.

**Pull request merged**